tests/core/creators/_helpers/tools.js

```diff
@@ -42,9 +44,37 @@ var detachedTests = ( function() {
 					delayIfDetached: true
 				} );
 
-				assert.isNull( editor, 'Editor should not be created on detached element, if config allows a delay. Creator function used: ' + creatorFunction );
+				assert.areSame( 'function', typeof editor, 'Editor should return function that allows to cancel creation. ' + assertMessage );
+
+				editorParent.append( editorElement );
+			},
+
+			test_delayed_editor_creation_is_cancelable: function() {
+				var editorElement = CKEDITOR.document.getById( createHtmlForEditor() ),
+					editorParent = editorElement.getParent();
+
+				editorElement.remove();
+
+				var cancelationCallback = CKEDITOR[ creatorFunction ]( editorElement, {
+					delayIfDetached: true,
+					delayIfDetached_interval: 50
+				} );
+
+				cancelationCallback();
+
+				// Attach the spy after cancelation to prevent unexpected invocations while test is waiting to fullfil.
+				var spyIsDetached = sinon.spy( editorElement, 'isDetached' );
+
+				CKEDITOR.tools.setTimeout( function() {
+					resume( function() {
+						assert.areSame( 0, spyIsDetached.callCount, 'The isDetached method should not be called after cancelation. ' + assertMessage );
+						spyIsDetached.restore();
+					} );
+				}, 150 );
 
 				editorParent.append( editorElement );
+
+				wait();
```

Comment:
That's still not totally valid test case. We should make sure that the editor is not initialized. Checking internal calls is not enough in that regard and we should make sure that the creation method has not been executed. It would be best to not use spy at all, but editor initialization takes such long that simplifying it with spy on `creatorFunction` should be enough.